test/unit/cute/intel_xe/CMakeLists.txt

```diff
@@ -26,17 +27,41 @@
 # OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-if(NOT DEFINED IGC_VERSION_MAJOR)
-  set(IGC_VERSION_MAJOR 2)
-endif()
-if(NOT DEFINED IGC_VERSION_MINOR)
-  set(IGC_VERSION_MINOR 18)
-endif()
+# Function to check IGC version
+function(check_igc_version)
+    execute_process(
+        COMMAND bash -c "dpkg -l | grep -E 'intel-igc-(core|opencl)(-[0-9]+)?' | awk '{print $3}' | cut -d'.' -f1,2 | head -1"
```

Comment:
This is not ideal.
- It won't work on some Linux distributions (e.g. RPM based)
- It will give the wrong result if the user has set the LD_LIBRARY_PATH to use a different IGC version which we commonly use for development.

It would be better to call `ocloc query OCL_DRIVER_VERSION`. Sadly this doesn't directly return the IGC version (GSD-11319). So this then still needs mapping or using the driver version for the macro.

This is still not ideal if the target is SPIR-V because then IGC is called at runtime and by that time the IGC might be different. So in that case it would be better to use a runtime query.

Reply:
will be done in follow up PR